### PR TITLE
Adding version details to the PWA

### PR DESCRIPTION
**Pull request by blckmn**
Not sure if this is the right way to go about it...

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by chmelevskij — APPROVED** [review]
Do we need this in gulpfile if this versioning is only used in web version?

Another way to do it would be to inject env variables into this the build in vite then use `import.meta.env` to read those out. That would remove need to create that extra file.

**Comment by blckmn** [comment]
> Do we need this in gulpfile if this versioning is only used in web version?
> 
> Another way to do it would be to inject env variables into this the build in vite then use `import.meta.env` to read those out. That would remove need to create that extra file.

Agree. I think the original idea was better, and simply needed the global variables (introduced by Vite definitions) to be marked as readonly.

**Comment by blckmn** [comment]
@haslinghuis will need someone with SonarCloud access to either give me access or review the security issue (doesn't like git on the path).

File: vite.config.js
Comment (haslinghuis) [inline vite.config.js:7]: Missing semicolon
```suggestion
import pkg from './package.json';
```

File: vite.config.js
Comment (haslinghuis) [inline vite.config.js:55]: This does not translate to path executable
```
C:\dev\betaflight-configurator>git rev-parse --short HEAD
d042471d
```

Comment (blckmn) [reply vite.config.js:55]: The issue is sonarcloud wants the full path to git. It doesn't like that I've let the system search for it... it can just be reviewed as 'safe'.

Comment (haslinghuis) [reply vite.config.js:55]: it's asking the path for git 🤕

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
seemingly caused a bug in Android builds.
https://discord.com/channels/868013470023548938/1174400159539544096/1174787103083733115
https://discord.com/channels/868013470023548938/996119401269047397/1180145077251735673

**Pull request merged**
